### PR TITLE
implement 3party

### PR DESCRIPTION
**Pull request by aofattaporn**
** implement google signIn**

- delete authorize bloc
- add event signInwithGoogle
- add state signInwithGoogle
- custom signIn UL check state from authentication bloc

### CODE REVIEW

**Comment by aofattaporn** [comment]
pls can you review my code @teamchz  @sura1542  @thunyathep  @Balloii

**Review by teamchz — APPROVED** [review]
LGTM

**Comment by aofattaporn** [comment]
> LGTM

that's so fast, I am waiting for another to see this review.

File: lib/src/data/respository/auth_provider.dart [hunk 1]
```diff
@@ -129,20 +132,19 @@ class AuthRepository {
...
-      }
+  Future<String> GetMyUser() async {
+    String? token = await FirebaseAuth.instance.currentUser?.getIdToken();
+    final response =
+        await http.get(Uri.parse("${globalUrl}v1/users/id"), headers: {
+      'Content-Type': 'application/json',
+      'Accept': 'application/json',
+      'Authorization': 'Bearer $token',
+    });
+    print(response.request);
```
Comment (thunyathep) [inline lib/src/data/respository/auth_provider.dart]: If you want to clean code I think you should delete print(response.request) ? In my opinion

**Review by thunyathep — APPROVED** [review]

**Pull request merged**